Natalie Rehfeld
Lab 2

Objectives: The goal of this Lab is to learn how to construct confidence intervals and to test hypotheses. Before proceeding with hypothesis tests, you have to understand the real-world process and how to formulate problems using the instructions described below. Visit the QE lab manual for all Minitab and Python codes for hypothesis testing.

Instructions: Three tests will be demonstrated: (A) tests for mean with variance unknown; (B) tests of hypothesis on the means of two normal distributions, variances unknown; (C) tests on variances of normal distributions. For Minitab instructions, please go to the lab manual. 

In [ ]:
from google.colab import drive

drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
import math
import statistics
import numpy as np
import scipy.stats
import pandas as pd
from statsmodels.stats import weightstats as stests
from scipy.stats import norm
import matplotlib.pyplot as plt

In [ ]:
data=pd.read_csv('drive/MyDrive/Lab2 Data.csv', sep=',', na_values=".")

In [ ]:
data

Specimen   VIQ
0          1  3193
1          2  3124
2          3  3153
3          4  3145
4          5  3093
5          6  3466
6          7  3355
7          8  2979
8          9  3182
9         10  3227
10        11  3256
11        12  3332
12        13  3204
13        14  3282
14        15  3170

In [ ]:
scipy.stats.ttest_1samp(data['VIQ'], 3200)

Ttest_1sampResult(statistic=0.353451624297281, pvalue=0.7290157143441449)

In [ ]:
scipy.stats.t.interval(.05,14,np.mean(data['VIQ']),np.std(data['VIQ']))

(3203.479572174935, 3217.987094491731)

To make the two sided test a one sided test, we would divide the P value in half
(1-.729016)*100=27%
The P value is 0.729, therefore, we cannot reject the null hypothesis.

In [ ]:
CF = 0.95
df = data.size-1
mean = np.mean(data)
std = scipy.stats.sem(data)

confidence_interval = scipy.stats.t.interval(CF, df, mean, std )

print(confidence_interval)

(array([   5.63837223, 3148.62545213]), array([  10.36162777, 3272.84121453]))


Part B

In [ ]:
data2 = pd.read_csv('drive/MyDrive/Lab2 Data2.csv', sep=',', na_values=".")
data2

Observation Number  Catalyst 1  Catalyst 2
0                   1       91.50       89.19
1                   2       94.18       90.95
2                   3       92.18       90.46
3                   4       95.39       93.21
4                   5       91.79       97.19
5                   6       89.07       97.04
6                   7       94.72       91.07
7                   8       89.21       92.75

In [ ]:
scipy.stats.ttest_ind(data2['Catalyst 1'], data2['Catalyst 2'], equal_var = True)

Ttest_indResult(statistic=-0.3535908643461798, pvalue=0.7289136186068217)

The hypotheses are H0: u1=u2 and H1: u1/=u2

This is a two sided test

Part C

In [ ]:
x = [74,76,74,69,58,71,66,65]
y = [78,79,75,66,63,70,66,67]

In [ ]:
def f_test(x, y):
  x = np.array(x)
  y = np.array(y)
  f = np.var(x, ddof=1)/np.var(y, ddof=1)
  dfn = x.size-1
  dfd = y.size-1
  p = 1-scipy.stats.f.cdf(f, dfn, dfd)
  return f, p


f_test (x,y)


(0.9646317829457366, 0.5183322097160306)

Since the p-values are greater than 0.5 we cannot reject the null hypothesis